org.mfek.glif.yml

```diff
@@ -0,0 +1,44 @@
+app-id: org.mfek.glif
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.rust-nightly
+build-options:
+  append-path: /usr/lib/sdk/rust-nightly/bin
+  env: 
+    CARGO_HOME: /run/build/MFEKglif/cargo
+    SKIA_BINARIES_URL: "file:///run/build/MFEKglif/skia-binaries.tar.gz"
+    FORCE_SKIA_BINARIES_DOWNLOAD: "1"
+finish-args: 
+  - --share=ipc
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --device=dri
+  - --filesystem=home
```

Comment:
Why do you need this? Does it support filechoosers?

Reply:
Yes, MFEKglif supports file choosers on all three platforms. On GNU/Linux, GTK3's one is used.

Reply:
> Yes, MFEKglif supports file choosers on all three platforms. On GNU/Linux, GTK3's one is used.

Do they use the portal https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.FileChooser.html ? If so, you don't need this.

Reply:
The portal is not used. So having filesystem access of some sort is needed to be able to use the native file choosers. Given the fact that programs like Blender use host filesystem access, I don't see why it's a problem to use this permission

Reply:
> Given the fact that programs like Blender use host filesystem

1. it's hard to revert without breaking
2. two wrongs don't make a right
3. this is the problem with exceptions: everybody use them as the reason to ask for one.

Reply:
If the UFO's guidelines change, MFEKglif changes the guidelines visible via Linux inotify, so the glif opened is never the only  file accessed. Layer metadata also accessed.

Reply:
(&, Ctrl-O reopens “Open” dialog, it's also under “File”.)

![Screenshot_20231128_230508](https://github.com/flathub/flathub/assets/838783/6c9e3943-228b-4b2c-bfa9-a1b0175a7bef)

How is this “wrong” in any application?

Reply:
> The portal is not used. 

You should switch to something that supports the portal. 

Right now seems like nfd-rs https://github.com/MFEK/glif/blob/53f452363e97d402a5dbc9a02ce0d3177909945e/src/filedialog.rs is used which provides bindings for https://github.com/mlabbe/nativefiledialog, and has stopped development/doesn't seem to support the portal.

Reply:
> You should switch to something that supports the portal.

Why? NFD works fine, and if I do this, then I am giving up the ability to customize the file dialog on all platforms, because as things stand I can replace nfd.rlib with a future e.g mfek-filedialog.rlib which would provide, for example, UFO-aware file-dialogs which don't show them as directories, like FontForge does.

Reply:
Am I right in thinking that “the rules” only apply to small projects?

Reply:
I can't force you to, hence "should". Portals are better than using static permissions. You have to decide the merits/demerits of it in the context of your app.

Reply:
Eg. Apart from getting access to any files/folders without a static filesystem permission, you do get DE-native filechoosers.

Nativefiledialogue seems to support only GTK and an optional zenity backend. I guess it'd look weird on KDE/Qt.